src/handlers/Resolver.ts

```diff
@@ -245,15 +251,18 @@ export async function handleVersionChanged({
   const { node } = event.args;
   const id = makeResolverId(event.log.address, node);
   const domain = await context.db.find(schema.domain, { id: node });
-  if (!domain) throw new Error("domain expected");
 
-  // materialize the Domain's resolvedAddress field
-  if (domain.resolverId === id) {
+  // materialize the Domain's resolvedAddress field iff exists and is set to this Resolver
```

Comment:
```suggestion
  // clear the Domain's resolvedAddress field iff the domain is already indexed and the domain's active Resolver is the Resolver emitting this event
```

There's more to the logic here, but cautious if I'm interpreting the next ideas correctly:

The key issue for improvement here is how we're documenting "Domain" here and how it relates to the Resolver that is emitting this event.

For example, if we review how `id` and `domain` are determined as a function of the `event`, it looks like this domain has a relation to the resolver emitting the event.

Trying to get this PR feedback sent to you quickly so won't investigate further right now.